rfcs/0045-zkapp-balance-data-in-archive.md

```diff
@@ -92,13 +88,33 @@ The table `balances` is replaced by a new table `accounts_accessed`, with column
   zkapp                   int               REFERENCES zkapp_accounts(id)
 ```
 
-In order to include the genesis ledger accounts in this table, we may need
-a separate app to populate it. Alternatively, we could use an app to dump the SQL needed
-to populate the table, and keep that SQL in the Mina repository.
+In order to include the genesis ledger accounts in this table, we may
+need a separate app to populate it. Alternatively, we could use an app
+to dump the SQL needed to populate the table, and keep that SQL in the
+Mina repository.
 
-The new table `zkapp_sequence_states` has the same definition as the existing `zkapp_states`;
-it represents a vector of field elements.  We probably don't want to commingle sequence states
-with app states in a single table, because they contain differing numbers of elements.
+The new table `account_ids`:
+```
+  id                 serial  PRIMARY_KEY
+  public_key_id      int     NOT NULL     REFERENCES public_keys(id)
+  token_id           int     NOT NULL     REFERENCES tokens(id)
+  token_owner        int                  REFERENCES account_ids(id)
+```
+A `NULL` entry for the `token_owner` indicates that this account
+owns the token.
+
+The new table `tokens` is:
```

Comment:
I think this suffers from the same issue: the token owner can update the `token_symbol` at any time (depending on how their zkApp is set up), and so we would need to update the `token_id` field of every member of `account_ids` that uses it. I would prefer if `token_symbol` stays inside the account and we keep just `token` in `account_ids`.